### PR TITLE
KFSPTS-20632 eInvoice getVendors API Call POST

### PR DESCRIPTION
**Pull request by ajd299**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/sys/web/service/impl/CuApiJsonWebRequestReaderImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import edu.cornell.kfs.sys.web.service.CuApiJsonWebRequestReader;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.BadRequestException;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public class CuApiJsonWebRequestReaderImpl implements CuApiJsonWebRequestReader {
```
Comment (ajd299) [inline src/main/java/edu/cornell/kfs/sys/web/service/impl/CuApiJsonWebRequestReaderImpl.java:15]: Once the review is passed I plan to create a separate user story to use this method in the CapAsset API.

File: src/main/java/edu/cornell/kfs/sys/web/service/impl/CuApiJsonWebRequestReaderImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	@Override
+    public JsonNode getJsonContentFromServletRequest(HttpServletRequest request) throws BadRequestException, IOException {
+        try (var requestInputStream = request.getInputStream();
+             var streamReader = new InputStreamReader(requestInputStream, StandardCharsets.UTF_8)) {
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/sys/web/service/impl/CuApiJsonWebRequestReaderImpl.java:20]: formatting seems a bit off here

File: src/main/java/edu/cornell/kfs/sys/web/service/impl/CuApiJsonWebRequestReaderImpl.java (hunk 1 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/sys/web/service/impl/CuApiJsonWebRequestReaderImpl.java:15]: sounds like a good plan. Will that require full regression testing of the CapAsset tool?

File: src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java [hunk 3]
```diff
@@ -316,6 +339,13 @@ private CuEinvoiceDao getCuEinvoiceDao() {
         return cuEinvoiceDao;
     }
 
+    private CuApiJsonWebRequestReader getCuApiJsonWebRequestReader() {
+        if (ObjectUtils.isNull(cuApiJsonWebRequestReader)) {
+            cuApiJsonWebRequestReader = SpringContext.getBean(CuApiJsonWebRequestReaderImpl.class);
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java:344]: should this be SpringContext.getBean(CuApiJsonWebRequestReader.class); ?

File: src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java [hunk 4]
```diff
@@ -160,13 +176,20 @@ public Response getUnitOfMeasureCodes(@Context HttpHeaders headers) {
...
 
-    private List<String> getVendorNumbersFromRequest() {
-        String vendorNumbersString = servletRequest.getParameter(CUPurapConstants.Einvoice.VENDOR_NUMBERS);
-        if (StringUtils.isEmpty(vendorNumbersString)) {
-            throw new BadRequestException();
+    private List<String> getVendorNumbers() throws IOException, BadRequestException {
+        String badRequestErrorMessage = "Invalid Request Body, expect JSON Object with \"vendors\" property of type array";
+        JsonNode jsonNode = getCuApiJsonWebRequestReader().getJsonContentFromServletRequest(servletRequest);
+
+        if (!jsonNode.has("vendors") || !jsonNode.get("vendors").isArray()) {
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java:183]: should there be a constant for "vendors"?

**Review by db483 — CHANGES_REQUESTED** [review]
Changes look good but I had a few questions and suggestions. Thank you.

**Review by db483 — APPROVED** [review]
Changes look good.

**Pull request merged**